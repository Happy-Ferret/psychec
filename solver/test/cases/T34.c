void f() {
    struct node a;
    a.x = 0; // pointer or integer?
    a.w = 5; // integer
    a.y = &a;
    *(a.z) = a;

    struct data b, *c;
    b.m = 0; // m classified as scalar
    b.m = c; // m grows higher in the lattice, classified as pointer
}
