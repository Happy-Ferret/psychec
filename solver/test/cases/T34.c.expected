/* Forward declarations */

/* Type definitions */
struct node {int w; struct node* z; struct node* y; int /*<<< orphan or scalar*/ x; } ;
struct data {struct data* m; } ;

/* Variables and functions*/
